### ISSUE TITLE
DP linear regression

### ISSUE THREAD

**Issue by LaRiffle** [arkhn/dsPrivacy#22]
## Problem

Linear regression can be solved by gradient descent, using the analogy of a 1 hidden layer neural network, see for instance : https://www.cs.cmu.edu/afs/cs.cmu.edu/academic/class/15381-s06/www/nn.pdf

Opacus is a Python library implementing neural network regression with DP-stochastic gradient descent

If we wrap Opacus in R, we should be able to propose an Linear regression solver with DP
